src/uri.cr

```diff
@@ -426,4 +426,91 @@ class URI
       URI.escape(password, io)
     end
   end
+
+  # A map of schemes and their respective default ports, seeded
+  # with some well-known schemes sourced from the IANA [Uniform
+  # Resource Identifier (URI) Schemes][1] and [Service Name and
+  # Transport Protocol Port Number Registry][2] via Mahmoud
+  # Hashemi's [scheme_port_map.json][3].
+  #
+  # [1]: https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml
+  # [2]: https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml
+  # [3]: https://gist.github.com/mahmoud/2fe281a8daaff26cfe9c15d2c5bf5c8b
+  @@default_ports = {
+    "acap"     => 674,
+    "afp"      => 548,
+    "dict"     => 2628,
+    "dns"      => 53,
+    "ftp"      => 21,
+    "ftps"     => 990,
+    "git"      => 9418,
+    "gopher"   => 70,
+    "http"     => 80,
+    "https"    => 443,
+    "imap"     => 143,
+    "ipp"      => 631,
+    "ipps"     => 631,
+    "irc"      => 194,
+    "ircs"     => 6697,
+    "ldap"     => 389,
+    "ldaps"    => 636,
+    "mms"      => 1755,
+    "msrp"     => 2855,
+    "mtqp"     => 1038,
+    "nfs"      => 111,
+    "nntp"     => 119,
+    "nntps"    => 563,
+    "pop"      => 110,
+    "prospero" => 1525,
+    "redis"    => 6379,
+    "rsync"    => 873,
+    "rtsp"     => 554,
+    "rtsps"    => 322,
+    "rtspu"    => 5005,
+    "scp"      => 22,
+    "sftp"     => 22,
+    "smb"      => 445,
+    "snmp"     => 161,
+    "ssh"      => 22,
+    "svn"      => 3690,
+    "telnet"   => 23,
+    "ventrilo" => 3784,
+    "vnc"      => 5900,
+    "wais"     => 210,
+    "ws"       => 80,
+    "wss"      => 443,
+  }
+
+  # Returns the default port for the given *scheme* if known,
+  # otherwise returns `nil`.
+  #
+  # ```
+  # URI.default_port "http"  # => 80
+  # URI.default_port "ponzi" # => nil
+  # ```
+  def self.default_port(scheme : String) : Int32?
```

Comment:
Since the `URI.default_port` method can return `nil`, should it be suffixed with a question mark (`URI.default_port?`) as per other methods such as `Hash#first_key?` that can return `nil`?

Reply:
No, this is only used when there is also a variant of the method which does not return `nil` (usually raises instead). This makes no sense here.

Reply:
Awesome, thanks very much for the explanation!